### PR TITLE
Adding support for Wildfly 31

### PR DESCRIPTION
**Pull request by febilus**
This is a fix for [6997](https://github.com/apache/netbeans/issues/6997)
With wildfly 31 the location of the MANIFEST.MF has changed. It's now in wildfly-feature-pack-product-conf-31.0.0.Final.jar

I created a method to extract the Manifest Attributes from the jar file.  Otherwise it ready the MANIFEST.MF as before.

### CODE REVIEW

File: enterprise/javaee.wildfly/src/org/netbeans/modules/javaee/wildfly/ide/ui/WildflyPluginUtils.java [hunk 1]
```diff
@@ -435,6 +431,29 @@ public static boolean isWildFlyJakartaEE(File serverPath) {
...
+        for (File file : files) {
+            try {
+                if (file.getName().startsWith("wildfly-feature-pack-product-conf") && file.getName().endsWith(".jar")) {
+                    JarFileSystem featurePackProductConfJar = new JarFileSystem(file);
+                    return featurePackProductConfJar.getManifest().getMainAttributes();
+                }
+            } catch (Exception ignore) {
+            }
+        }
+        File manifestFile = new File(serverPath, getModulesBase(serverPath.getAbsolutePath()) + "org.jboss.as.product".replace('.', separatorChar) + separatorChar + slot + separatorChar + "dir" + separatorChar + "META-INF" + separatorChar + "MANIFEST.MF");
```
Comment (matthiasblaesing) [inline enterprise/javaee.wildfly/src/org/netbeans/modules/javaee/wildfly/ide/ui/WildflyPluginUtils.java:446]:
```suggestion
        File manifestFile = new File(productDir, "dir" + separatorChar + "META-INF" + separatorChar + "MANIFEST.MF");
```

**Review by matthiasblaesing — APPROVED** [review]
Change looks good to me. I checked the Wildfly 31 download and found the structure this change is based on. I have a minimal change suggestion to make it easier to read.

**Comment by mbien** [comment]
@febilus could you squash all commits and force push into the PR?

**Comment by febilus** [comment]
Hi @mbien , I got a question, because I think I did a small mistake.
After  I took  your suggestion I did a mistake: I updated my master and merged the master in the wildfly branch. Then I was not able to squash the last 2 commits together.

To resolve this I reset the merge with the master and squashed the changes together.
 
Now I'm not sure if I the PR is ready to use it, or do I have to update and merge the master into my branch again?

**Comment by mbien** [comment]
hi @febilus, you did good since I see only one commit in your branch - so the squashing worked.

the branch is a few weeks old but I think this shouldn't matter here since the Wildfly area has low traffic.

in future: lets say the PR is reviewed and someone asks you to squash the PR to make it ready to integration you can do the following:
you could update your branch (rewrite history) and put your commits on top of latest changes **without a merge commit**, all you have to do is:
```
git pull --rebase git@github.com:apache/netbeans master
```
(this assumes you have the PR branch checked out)
(you can also use origin alias if it points at the apache repo)
(this will also get rid of any existing merge commits, if there are any)

Now its easy to squash the commits however you like them to be (in most cases it will be just one commit).


Also keep in mind other projects might have a different policy, some don't allow force pushes and let the person who integrates the PR squash it etc.

**Comment by mbien** [comment]
tests are green, PR is reviewed and squashed -> merging

@febilus congrats to your first contribution!

**Comment by febilus** [comment]
Many Thanks for the explanation/help.

**Comment by SimonePadovan** [comment]
When will this fix be available ? thanks

**Comment by troizet** [comment]
This fix will be available in Apache NetBeans 22.

**Comment by mbien** [comment]
dev builds for the next release are also linked from the readme: https://github.com/apache/netbeans?tab=readme-ov-file#download

**Comment by SimonePadovan** [comment]
thank you very much !

**Comment by morvael** [comment]
Too bad such things can't be released via Netbeans Update mechanism.

**Comment by mbien** [comment]
@morvael you could give the dev builds a try

**Comment by morvael** [comment]
To test it or just switch for production work as workaround?

**Comment by mrcasio** [comment]
Thank you for the work on this PR! Unfortunately, it seems that this solution does not work for subsequent versions of WildFly. As mentioned in the Apache mailing list (source: https://lists.apache.org/thread/53k67ndngnf1fobfo1jzmo3cqb5ny3m2), the issue has resurfaced with WildFly 32 and WildFly 33. Specifically, NetBeans 22 looks for `MANIFEST.MF` in the `wildfly-feature-pack-product-conf-31.0.0.Final.jar`, which works for WildFly 31 but fails for later versions due to the file-name changes (e.g., `wildfly-feature-pack-product-conf-33.0.0.Final.jar` for WildFly 33). The user who reported this issue hopes for a similar solution to PR #7055, potentially for NetBeans 23, to enable compatibility with WildFly 33, which supports Java 21.

**Comment by mbien** [comment]
the manifest issue will likely by fixed by https://github.com/apache/netbeans/pull/7772, if this goes beyond the changed manifest name, please file a separate issue

**Pull request merged**